server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -157,6 +159,16 @@ static SearchRequest subSearchRequest(SearchRequest originalSearchRequest, Strin
         return new SearchRequest(originalSearchRequest, indices, clusterAlias, absoluteStartMillis, finalReduce);
     }
 
+    public static SearchRequest withMinimumVersion(SearchRequest searchRequest, Version minVersion) {
```

Comment:
Should we add a check that the min version is greater or equals than the minimum compatibility version ?

Reply:
We should also ensure that `ccsMinimizeRoundtrips` is set to false, otherwise the check will not be applied on the remote cluster. We can force the value here `searchRequest.cssMinimizeRountrips(false)` but we should also verify the final value in `SearchRequest#validate`.